traitsui/testing/tester/wx/implementation/editor_factory_readonly_editor.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+import wx
+
+from traitsui.wx.editor_factory import ReadonlyEditor
+from traitsui.testing.tester import query
+
+
+def readonly_displayed_text_handler(wrapper, interaction):
+    ''' Handler for ReadonlyEditor to handle query.DisplayedText interactions.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+        the UIWrapper object wrapping the ReadonlyEditor
+    interaction : Instance of query.DisplayedText
+        This arguiment is not used by this function. It is included so that
+        the function matches the standard format for a handler.  The intended
+        interaction should always be query.DisplayedText
+
+    Notes
+    -----
+    wx Readonly Editors occassionally use wx.TextCtrl as their control, and
+    other times use wx.StaticText.
+    '''
+    if isinstance(wrapper.target.control, wx.TextCtrl):
+        return wrapper.target.control.GetValue()
+    elif isinstance(wrapper.target.control, wx.StaticText):
+        return wrapper.target.control.GetLabel()
+    raise TypeError("readonly_displayed_text_handler expected a UIWrapper with"
+                    " a ReadonlyEditor as a target. ReadonlyEditor control"
+                    " should always be either wx.TextCtrl, or wx.StaticText."
+                    " {} was found".format(wrapper.target.control))
```

Comment:
I think I have seen this function before that is close to identical. Is it time to refactor?

Reply:
It is exactly identical (even up to the name) as the function in the text_editor implementation.  Yes, I agree it should be pulled out somewhere and used in both places.  I'm unsure where though.  Of the current options maybe the wx `registry_helper` makes most sense?  It could also go in its own new module (but we do have a lot of modules with only one function or class in them already).  What do you think?

Reply:
Since this function only uses `wrapper.target.control` and does not use `interaction`, it might fit in the more `helpers` module instead?

Reply:
Yeah I think i agree.  I originally had thought helpers seemed incorrect as currently those functions are all directly related to mouse clicks/ key clicks, but they are basically just functions that are called when `.perform()` is called.  It makes sense that this would be there as well as it will be called when `.inspect()` is called.